libr/core/linux_heap_glibc.c

```diff
@@ -21,6 +21,37 @@
 #define GHT_MAX UT64_MAX
 #endif
 
+
+static GHT GH(get_va_symbol)(RCore *core, const char *path, const char *symname) {
+	RBin *bin = core->bin;
+	GHT vaddr = GHT_MAX;
+	RListIter *iter;
+	RBinSymbol *s;
+	const char *bin_file = bin->file;
+	ut64 baddr = r_config_get_i (core->config, "bin.baddr");
+	ut64 laddr = r_config_get_i (core->config, "bin.laddr");
+	int fd = r_io_fd_get_current (core->io);
+	int rawstr = bin->rawstr;
+
+	RBinOptions opt;
+	r_bin_options_init (&opt, -1, 0, 0, false);
+	r_bin_open (bin, path, &opt);
```

Comment:
@trufae @meowmeowxw This use of RBin looks wrong IMHO. You end up keeping the binary file for libc opened, moreover you then re-open the binary under analysis, but there is already a binfile opened for it.

If you do `dmh` and then `ob`, you can see the opened binary files:
```
1 3 x86-64 ba:0x556f0b9c8000 sz:144970 /usr/bin/ls
0 4 x86-64 ba:0x00000000 sz:3178154 /usr/lib64/libc-2.30.so
2 3 x86-64 ba:0x556f0b9c8000 sz:144970 /usr/bin/ls
```

Reply:
Do you suggests any alternative? I was using `r_bin_new` and then `r_bin_free` to close the bin but it forced the loading of the plugin twice: https://github.com/radareorg/radare2/pull/17018#discussion_r436851925

Reply:
i didn't forced you :P just suggested that creating new instances is unnecessary payload when the plugins and libs are already there. you are calling RBin.open() twice, but never calling RBin.close(), this is why its leaking files

Reply:
I think it should probably look something like:
```C
RBinFile *current_bf = r_bin_cur (bin);
r_bin_open (bin, path, &opt);
// get symbols as done now
RBinFile *libc_bf = r_bin_cur (bin);
r_bin_file_close(bin, bf->id);
r_bin_file_set_obj (bin, bf, bf->o);
```

I have not tested it, but unless there are bugs somewhere, I think these are the API we should use.